File: docs/algorithms/code/algorithms/twoSum1.ts
function twoSum(price: number[], target: number): number[] {
  let left = 0,
    right = price.length - 1;

  while (left < right) {
    const sum = price[left] + price[right];
    if (sum === target) {
      return [price[left], price[right]];
    }

    if (sum > target) {
      right--;
    } else {
      left++;
    }
  }

  return [-1, -1];
}

File: docs/algorithms/code/algorithms/twoSum1.go
func twoSum(nums []int, target int) []int {
	n := len(nums)
	if n == 0 {
		return []int{}
	}

	hashTable := make(map[int]int)

	for i, x := range nums {
		p, ok := hashTable[target-x]

		if ok {
			return []int{p, i}
		} else {
			hashTable[x] = i
		}
	}

	return []int{}
}
